adrs/0003-testing-strategy.md

```diff
@@ -0,0 +1,49 @@
+# Stacks Testing Strategy
+
+**Date:** 2022-11-28
+
+**Related PR:** [#1195](https://github.com/StackExchange/Stacks/pull/1195)
+
+## Issue
+We currently don't test our Stacks components except from manually/ad-hoc executing some visual regression tests of entire documentation pages with BackstopJS. Linting and formatting despite being setup are also often ignored because they are not running methodically in a pipeline.
+
+**A lack of testing gives us very little confidence refactoring and evolving the codebase.**
+
+## Decision
+
+Introduce 2 initial categories of tests for Stacks to increase confidence working on new and existing components:
+- Component/Unit Tests
+- Component Visual Regression Tests
+
+General guidelines:
+- Tests run in a headless browser context ([wtr + playwright](https://modern-web.dev/docs/test-runner/overview/)) to provide higher level of confidence
+- DOM Querying and Events Simulation are done as much as possible with [testing-library](https://testing-library.com/) and [user-event](https://testing-library.com/docs/user-event/intro/) instead of reaching out directly to DOM APIs
+- Every component should be accompanied by related component and visual regression tests (atomic classes are excluded)
+
+### Component/Unit Tests
+These tests should provide full coverage of the component being tested. Happy path, edge and error cases should be all covered.
+
+Guidelines:
+- Tests should not reach for internal implementation/methods but rather tests the component through its public APIs (from the point of view of a consumer)
+- Complex DOM-agnostic business logic can be extracted in separate functions which can be unit tested in isolation
+
+### Component Visual Regression Tests
+These tests should check how the component will look after any code changes have been made by comparing screenshots taken before and after code changes. 
+
+Guidelines:
+- Baseline images for these tests should be committed to the repository with `git lfs` to avoid bloating the history.
+- When relevant, screenshots at different viewport sizes should be taken
+
+### Additional Guidelines
+
+- All tests and linters runs in a GH workflow for every commit on any branch.
+
+- BackstopJS tests will be gradually retired in favor of the more targeted component visual regression tests approach.
```

Comment:
I don't think we actually have any backstop tests (and if we do, I don't think I've ever run them), so I don't see any reason we can't change this to "immediately retire[d]"

Reply:
I run them manually sometimes while reviewing PRs which should not have any visual changes (usefulness: low - lot of false positives cluttering the output). So I would be happy to get rid of them straight away. 
I also saw @dancormier running them though - @dancormier would you feel comfortable ditching backstop straight away? As I said in my other comment if we want to keep snapshotting entire documentation pages perhaps we could look into mabl (or just use playwright again 😀).

Reply:
Honestly, BackstopJS is so noisy that I've recently resorted to doing manual image diffing of docs pages. I'm good with ditching backstop straight away.

Reply:
Alright I will update both PRs (ADR and implementation) to ditch BackstopJS straight away.